### ISSUE TITLE
Thread and memory leak when spawning/terminating

### ISSUE THREAD

**Issue by sveniu** [wez/wezterm#6116]
### What Operating System(s) are you seeing this problem on?

Linux Wayland

### Which Wayland compositor or X11 Window manager(s) are you using?

Gnome 46 on Ubuntu 24.04.1.

### WezTerm version

30345b36d8a00fed347e4df5dadd83915a7693fb

### Did you try the latest nightly build to see if the issue is better (or worse!) than your current version?

Yes, and I updated the version box above to show the version of the nightly that I tried

### Describe the bug

(The version indicates the main branch commit hash as of 2024-Sep-10. I cloned the repo to build locally, to get a binary with debug symbols.)

When spawning/terminating terminal windows that reuse the single wezterm-gui instance, each spawn/terminate cycle increases memory use by around 5 MB, and adds two threads. Neither seem to disappear over time.

I've often seen memory usage ballooning to gigabytes over multi-day working sessions.

### To Reproduce

1.  Start the main wezterm instance: `wezterm`
2. In the spawned window, run a loop: `for i in $(seq 1 200); do wezterm start -- sh -c 'exit 0'; done`
3. Observe memory use (RSS) starting at ~300 MB and ending around 1.2 GB; ~5 MB per cycle.
4. Observe thread count starting at 17 and ending at 417; two per cycle.

### Configuration

no config

### Expected Behavior

Memory use stays around ~300 MB as when the main instance first started, of course with increases here and there due to scrollback and such.

Massif output and gdb `info threads` attached below. Seems that some dependencies lack symbols?

### Logs

[massif.out.txt](https://github.com/user-attachments/files/16950109/massif.out.txt)
[wezterm-gdb-info-threads.txt](https://github.com/user-attachments/files/16950125/wezterm-gdb-info-threads.txt) – note that this is after only 20 cycles of spawn/terminate (not 200).

**Comment by wez** [comment]
The thread names are not for threads created directly by wezterm. I suspect that they are related to your graphics card drivers.
You might consider changing https://wezfurlong.org/wezterm/config/lua/config/front_end.html to try a slightly different set of drivers, upgrading to the latest `main` from today (I just updated the wgpu dependency, which may be relevant here), or looking at your graphics card drivers themselves.

**Comment by sveniu** [comment]
I've ended up with `enable_wayland = false` to work around the issue for now.

This might be the same issue as #4532, given I saw an error "warning: queue 0x... destroyed while proxies still attached" with references to wl_surface and such, in one of the test sessions. Seems likely to be an issue with Wayland, then.

I pulled 2d0c5cdd and rebuilt. Issue persists for any of `front_end = Software/OpenGL/WebGpu`. Also, setting

```
[profile.release]
opt-level = 1
debug = 2
```

in Cargo.toml did not give further symbols or insight into the Massif output; it still stops at `glium::program::raw::RawProgram::from_shaders`.

I guess it's glium that should do wl_..._destroy, and not Wezterm? Saw a similar issue here, and was wondering: https://gitlab.freedesktop.org/gstreamer/gstreamer/-/merge_requests/5219/diffs

I'll update Wayland/Gnome/drivers as they are made available in Ubuntu 24.04.1, and will update this issue with info later.
